client/scripts/views/components/search_bar.ts

```diff
@@ -225,295 +269,352 @@ const getBalancedContentListing = (unfilteredResults: any[], types: SearchType[]
+    const isMobile = window.innerWidth < 767.98;
+
+    const setFilterMenuActive = (using: boolean) => {
+      vnode.state.filterMenuActive = using
+    }
+
+    vnode.state.closeResults = () => {
+      vnode.state.hideResults = true;
+    };
 
-    const { results, searchTerm } = vnode.state;
-    const showDropdownPreview = !m.route.get().includes('/search?q=');
-    const isMobile = (window.innerWidth < 767.98);
-    const LoadingPreview = m(List, {
-      class: 'search-results-loading'
-    }, [ m(ListItem, { label: m(Spinner, { active: true }) }) ]);
-    const searchResults = (!results || results?.length === 0)
-      ? (app.searchCache[searchTerm]?.loaded)
-        ? m(List, [ m(emptySearchPreview, { searchTerm }) ])
-        : LoadingPreview
-      : vnode.state.isTyping
-        ? LoadingPreview
-        : m(List, { class: 'search-results-list' }, results);
-    vnode.state.closeResults = () => { vnode.state.hideResults = true; };
+    vnode.state.setUsingFilterMenu = using => {
+      vnode.state.filterMenuActive = using
+    }
+
+    const executeSearch = (query: SearchQuery) => {
+      if (
+        !query.searchTerm ||
+        !query.searchTerm.toString().trim() ||
+        !query.searchTerm.match(/[A-Za-z]+/)
+      ) {
+        notifyError('Enter a valid search term');
+        return;
+      }
+      if (query.searchTerm.length < 4) {
+        notifyError('Query must be at least 4 characters');
+      }
+      query.isSearchPreview = false;
+      app.search.addToHistory(query)
+      m.route.set(`/search?${query.toUrlParams()}`);
+    }
+
+    const historyList = app.search.getHistory()
+      .map(h => getSearchHistoryPreview(h, setFilterMenuActive, executeSearch))
+
+
+    if(historyList.length > 0) {
+      historyList.push(
+        m(ListItem, {class: 'search-history-no-results upper-border',
+          label: 'Tip: You can use operators like \'single quotes\', and the keyword "or" to limit your search!'})
+      )
+    }
+
+    const activeCommunity = app.community ? app.community.name : vnode.state.searchQuery.communityScope
+    const activeChain = app.activeChainId() || vnode.state.searchQuery.chainScope
+    const scopeTitle = m(ListItem, {class: 'disabled', label: 'Scope'})
+
+    const scopeButtons = [SearchScope.Threads, SearchScope.Replies, SearchScope.Communities, SearchScope.Members]
+      .map(s => {
+        return m(Button, {
+          size: Size.LG,
+          active: vnode.state.searchQuery.searchScope.includes(s),
+          onclick: () => {vnode.state.searchQuery.toggleScope(s)},
+          onmouseover: () => {vnode.state.filterMenuActive = true},
+          onmouseout: () => {vnode.state.filterMenuActive = false},
+          label: s
+        })
+      })
+
+    const filterDropdown =
+      m(List, {
+        class: 'search-results-list',
+      }, [
+        activeCommunity ? [scopeTitle, m(ListItem, {
+          class: 'disabled',
+          label: m(Button, {
+            size: Size.LG,
```

Comment:
Ditto on button sizing, maybe too big here

Would love to get Adam J's thoughts on this—one question is, "Do we need the header text, e.g. 'scope', 'I'm looking for', or is this context self-explanatory and we can just have button rows?" The other question is re: button sizing

![image](https://user-images.githubusercontent.com/22281010/144290072-8d52473b-452a-44af-8b32-e8552d5b24c6.png)

Reply:
I feel that if a search is being inititated from a community page there should be  scope filter selected by default. 

The buttons in the screenshot yo uploaded have very wide left and right margins. Also they should be gray filled like in the design. I guess it's not done completely though.

There is a slight distinction between content filters and scope . filters so they should reside within separate rows. I scaled down the section titles . The original ones were very fat and tall.

Last comment is that in the results, the string should be highlighted. Like, the first instance of that string should be shown. So if the first instance happens in the 3rd paragraph, line 2, we should be showing line 2 and that string text with a background highlight of #C7B9EF. 

What do ya think?
![discussion threads](https://user-images.githubusercontent.com/80526644/144329631-16fddd47-0df5-4883-9e53-90e754deca29.png)
![search suggestions](https://user-images.githubusercontent.com/80526644/144329640-966fa386-66fd-4649-8a72-991f54c75e70.png)

Reply:
I restyled the buttons, and rearranged/restyled the menu to be much closer to the mocks. Let me know what you think @7racker